Spigot-Server-Patches/0211-Properly-handle-async-calls-to-restart-the-server.patch

```diff
@@ -15,28 +15,114 @@ restart command, and adds separate handling for async calls, such as those from
 the watchdog thread.
 
 When calling from the watchdog thread, we cannot assume the main thread is in a
-tickable state; it may be completely deadlocked. Therefore, we kill that thread
-right then and there.
+tickable state; it may be completely deadlocked. In order to handle this, we mark
+the server as stopping, in order to account for situations where the server should
+complete a tick reasonbly soon, i.e. 99% of cases.
+
+Should the server not enter a state where it is stopping, or; should the server be
+stuck stopping for 20 seconds, we assume that the server has deadlocked and
```

Comment:
Updated the patches commit message to reflect the actual current behavior, as well as note the decision